wordpress-mobile/WordPress-Android
# AppBar refresh - Stats, Pages, Posts, Media Browser and Domain Registration

This PR updates AppBar design and adds lift on scroll functionality to the following screens (and related nested screens):

- Stats
- Pages
- Posts
- Media Browser
- Domain Registration (only scrollable screen is contact details form).

What is lift on scroll? When you scroll content below the AppBar its elevation changes, and as a result shadow appears in light mode, and in Dark Mode color of AppBar gets lighter.

@osullivanchris I would like to get your feedback on abovementioned screens, and mostly on Stats :)

[![Image from Gyazo](https://i.gyazo.com/98af9d5d00364aca9eae816c0b8097ad.png)](https://gyazo.com/98af9d5d00364aca9eae816c0b8097ad)

To test:

For all screens:

- Make sure nothing looks out of place (including ActionMode) :)
- Make sure the toolbar state is restored on configuration change.

Test 1:

- Navigate to the Stats screen
- Make sure that lift on scroll works with all the tabs and switching between tabs preserves lift on scroll state.
- Make sure that lift on scroll works in Insights Management, View All, and Details screens.

Test 2:

- Navigate to Pages screen.
- Make sure that lift on scroll works with all the tabs and switching between tabs preserves lift on scroll state.
- Make sure that lift on scroll works with Search.

Test 3:

- Navigate to Posts screen.
- Make sure that lift on scroll works with all the tabs and switching between tabs preserves lift on scroll state.
- Make sure that lift on scroll works with Search.

Test 4:

- Navigate to the Media Browser screen.
- Make sure that lift on scroll works with all the tabs and switching between tabs preserves lift on scroll state.

Test 5:

- Using a site with unredeemed domain credit navigate to Domain Registration flow.
- Domain suggestions screen has an inner scroll view which should not trigger lift on scroll.
- Select a domain name from suggestion and proceed to contact info form.
- Make sure lift on scroll works as expected.

PR submission checklist:
